### ISSUE TITLE
Some songs just return a 403 upon being downloaded

### ISSUE THREAD

**Issue by milos192** [kmille/deezer-downloader#57]
Platform: Windows with the WSL2-backend for Docker (using Ubuntu as the distribution)
Build: latest as of 15.04.2021.

So, for the most part, the downloading goes fine, but there are some songs that straight up refuse to be downloaded and I just get a `ERROR: Can not download this song. Got a 403` every time I try. On the other hand, when downloading playlists, there are songs that fail at first, but work fine later on.

Here's a couple of IDs of songs that fail constantly: `9125239`, `8717671`, `8454029`.

The songs do play when I try them on the platform (both the web and Electron app).

Do you know why this might be? Let me know if I can provide more info, and thank you for the awesome app!

**Comment by kmille** [comment]
Hey,
all three examples for me: `9125239`, `8717671`, `8454029`. Example output:

Backend
```bash
User request: /download with {'type': 'track', 'music_id': 9125239, 'add_to_playlist': False, 'create_zip': False}
127.0.0.1 - - [20/Apr/2021 18:10:56] "POST /download HTTP/1.1" 200 -
Worker 1 is now working on task: {'track_id': 9125239, 'add_to_playlist': False}
127.0.0.1 - - [20/Apr/2021 18:10:56] "GET /queue HTTP/1.1" 200 -
Downloading 'Lazlo Bane - Superman.mp3'
Dowload finished: /tmp/deezer-downloader/songs/Lazlo Bane - Superman.mp3
```

Client
```bash
curl 'http://localhost:5000/download' --data-raw '{"type":"track","music_id":9125239,"add_to_playlist":false,"create_zip":false}'
```
Can you share your output?

**Comment by milos192** [comment]
I tried your exact command (copied and pasted).

Output:

```
User request: /download with {'type': 'track', 'music_id': 9125239, 'add_to_playlist': False, 'create_zip': False}
172.17.0.1 - - [20/Apr/2021 21:40:55] "POST /download HTTP/1.1" 200 -
Worker 1 is now working on task: {'track_id': 9125239, 'add_to_playlist': False}
Downloading 'Lazlo Bane - Superman.flac'
ERROR: Can not download this song. Got a 403
worker 1 is done with task: {'track_id': 9125239, 'add_to_playlist': False} (state=mission accomplished)
Worker 1 is waiting for a task
```

Can you try with FLAC to see if it will work then?

Thanks for the response!

**Comment by kmille** [comment]
hmm this works for me too
```bash
User request: /download with {'type': 'track', 'music_id': 65445466, 'add_to_playlist': False, 'create_zip': False}
127.0.0.1 - - [21/Apr/2021 10:39:44] "POST /download HTTP/1.1" 200 -
Worker 1 is now working on task: {'track_id': 65445466, 'add_to_playlist': False}
Downloading 'The Doors - Riders on the Storm.flac'
Dowload finished: /tmp/deezer-downloader/songs/The Doors - Riders on the Storm.flac
worker 1 is done with task: {'track_id': 65445466, 'add_to_playlist': False} (state=mission accomplished)
```

hmm... Some thoughts:
1) maybe it's a legal issue and some songs are just available in certain regions? Can you play the song in the browser? https://www.deezer.com/us/track/9125239
2) Can you run it natively on Linux? 
3) Can you send me your cookie and I can try to download it (email is in `git log`)?

**Comment by milos192** [comment]
1. Nope, all the songs are available in the browser, and as a matter of fact, I've managed to download several songs that are region locked for me
2. If all else fails, I guess I could spin up a virtual machine, but I don't have Linux running on anything currently
3. Wrote you an email. Hope the address is correct. If not, let me know :)
4. You have downloaded a different song, not one of the IDs I mentioned; maybe it would not work with some of my IDs?

Thanks again!

**Comment by milos192** [comment]
Update: I started up my instance with the option to download MP3s and now the songs are downloading. Can you please try with FLAC quality on one of the three IDs I provided?

**Comment by kmille** [comment]
I already did this. See my last comment.

**Comment by milos192** [comment]
Hey, yeah, but I'm asking because the song in your console excerpt is Riders on the Storm, and not Superman.

Did you also try with the IDs I provided but just didn't paste that log?

Also, I think I got the email wrong, I'll try sending another one today.

Thanks again!

**Comment by kmille** [comment]
Hey,

seems like I haven't tested it right. I can reproduce the issue now (with id 9175916 and my cookie). The mp3 download works, flac doesn't. I can't tell you why. The http body is just empty. I have a premium subscription, but not the one with "HiFi" aka flac. So I can't tell you if it works for that song in the browser. 
Maybe someone can test this. There is nothing else I can do right now.

**Comment by milos192** [comment]
Thanks for the info! I guess all we can do is wait for someone with a HiFi subscription to try this out. Getting the MP3s will suffice for now. :)

**Comment by fweens** [comment]
Hi there ! Got the same problem with some songs... After reading this thread I've tried dowloading the MP3s ; it worked for some songs but some still don't work :-(

**Comment by milos192** [comment]
Can you give me an ID of a failing MP3 song? I'll try to get it to see if it's repro on my side, too.

**Comment by fsneew** [comment]
For instance : 468663282, 1199489112, 78592054, 886420052

**Comment by milos192** [comment]
Sorry for the very late reply. I managed to download 468663282, for instance. Didn't try the others as the first one worked.

How did you set up the container for MP3 downloading? Can you share the command you ran (without your ARL)?

**Comment by fsneew** [comment]
First I call : parse_deezer_playlist to get items of my playlist (and the ids) then I call get_song_infos_from_deezer_website(TYPE_TRACK,id) and this gives me the 404

**Comment by milos192** [comment]
Oh, sorry, I was just using the UI. It works fine for me from there.

**Comment by fsneew** [comment]
I found a workaround... Sometimes, for some reason, the ID returned by get_song_infos_from_deezer_website doesn't work (403 error) ; in this case I use the song["FALLBACK"]["SNG_ID"] ; still some 403 error on some songs but in progress...

**Comment by mostm** [comment]
> Thanks for the info! I guess all we can do is wait for someone with a HiFi subscription to try this out. Getting the MP3s will suffice for now. :)

Trying with HiFi subscription account, issues are same.

**Comment by Mushebeq** [comment]
literally every song i request is getting this error

**Comment by kmille** [comment]
Please check #66 for further information
TLDR: right now, with a free subscription, downloading 128kbit/s mp3s should work (some 403s may occur).

**Comment by kmille** [comment]
After updating the download logic/API endpoints: Is this till an issue?

**Comment by milos192** [comment]
Hello there @kmille. I'm really sorry for the late response, you know how life goes sometimes - you just get lost in it for a few weeks.

These are my findings:
- 320 mp3 seems to be working fine for all tracks, except the ones that are region locked
- flac still does not work for a track from my original post (NOTE: I do not have the HiFi subscription anymore, just the regular Premium one)

Here's a log for a 320 mp3 that's not available in my country

```
User request: /download with {'type': 'track', 'music_id': 2230957397, 'add_to_playlist': False, 'create_zip': False}
Downloading 'Tommy heavenly6 - Pray (TV Version).mp3'
Could not download song (https://www.deezer.com/us/track/2230957397). Maybe it's not available anymore or at least not in your country. Could not get download url from API: Track token has no sufficient rights on requested media.
```

Here's a log for getting a FLAC track where a 320 mp3 is available and working (NOTE: I do not have the HiFi subscription anymore, just the regular Premium one)

```
User request: /download with {'type': 'track', 'music_id': 9125239, 'add_to_playlist': False, 'create_zip': False}
Downloading 'Lazlo Bane - Superman.flac'
Could not download song (https://www.deezer.com/us/track/9125239). Maybe it's not available anymore or at least not in your country. list index out of range
Task download_deezer_song_and_queue failed with parameters '{'track_id': 9125239, 'add_to_playlist': False}'
Reason: list index out of range
```

Hope this helps!

**Comment by Mushebeq** [comment]
I
